packages/fetch/src/types.ts

```diff
@@ -16,182 +16,174 @@
 }
 
-export type SuccessCallback = (response: CallbackResponse) => void;
+export { AbortSignal as AbortSignal };
 
-export type ErrorCallback = (err: Error) => void;
+declare class AbortController<PlatformAbortSignal = AbortSignal> {
+  /**
+   * Returns the AbortSignal object associated with this object.
+   */
+  readonly signal: PlatformAbortSignal;
 
-export interface ResponseHandler {
-  onSuccess: SuccessCallback;
-  onError: ErrorCallback;
+  /**
+   * Invoking this method will set this object's AbortSignal's aborted flag and signal to any observers that the associated activity is to be aborted.
+   */
+  abort(): void;
 }
 
-export interface NetworkTransport {
-  fetch<RequestBody>(request: Request<RequestBody>): Promise<FetchResponse>;
-  fetchWithCallbacks<RequestBody>(request: Request<RequestBody>, handler: ResponseHandler): void;
+export { AbortController };
+
+export declare class Headers {
+  append(name: string, value: string): void;
+  delete(name: string): void;
+  get(name: string): string | null;
+  has(name: string): boolean;
+  set(name: string, value: string): void;
+  forEach(callbackfn: (value: string, key: string, parent: this) => void): void;
+  getSetCookie?(): string[];
+  keys?(): IterableIterator<string>;
+  values?(): IterableIterator<string>;
+  entries?(): IterableIterator<[string, string]>;
+  [Symbol.iterator]?(): Iterator<[string, string]>;
 }
 
-// AbortController
+export type Blob = unknown;
+export type FormData = unknown;
+
+export declare class Response<PlatformHeaders extends Headers = Headers> {
+  // constructor (body?: BodyInit, init?: ResponseInit);
 
-type AbortSignal = unknown;
+  readonly headers: PlatformHeaders;
+  readonly ok: boolean;
+  readonly status: number;
+  readonly statusText: string;
+  readonly type: ResponseType;
+  readonly url: string;
+  readonly redirected: boolean;
 
-/** A controller object that allows you to abort one or more DOM requests as and when desired. */
+  readonly body?: ReadableStream | null;
+  readonly bodyUsed: boolean;
 
-export type AbortController = {
-  /**
-   * The prototype of an instance is the class.
-   */
-  prototype: AbortController;
+  readonly arrayBuffer: () => Promise<ArrayBuffer>;
+  readonly blob: () => Promise<Blob>;
+  readonly formData: () => Promise<FormData>;
+  readonly json: () => Promise<unknown>;
+  readonly text: () => Promise<string>;
 
-  /**
-   * Constructs an AbortController.
-   */
-  new (): AbortController;
+  readonly clone: () => this;
 
-  /**
-   * Returns the AbortSignal object associated with this object.
-   */
-  readonly signal: AbortSignal;
-  /**
-   * Invoking this method will set this object's AbortSignal's aborted flag and signal to any observers that the associated activity is to be aborted.
-   */
-  abort(): void;
-};
+  /*
+  static error(): Response;
+  static json(data: any, init?: ResponseInit): Response;
+  static redirect(url: string | URL, status: ResponseRedirectStatus): Response;
```

Comment:
Remove this?

Reply:
I'd like to keep it in to signal parts of the spec, which is explicitly left out.

Reply:
Perhaps a comment explaining that would be helpful

Reply:
I've comitted a comment linking to the docs.